### ISSUE TITLE
[Profile Page | Desktop] Components Implementation

### ISSUE THREAD

**Issue by ajmnz** [benawad/dogehouse#2022]
# Components Implementation

### Profile Page | Desktop

New components available for anyone that wants to contribute to the new design (`kibbeh`). Make sure to read the guidelines at the bottom of the issue.

---

### New components
- [x] ProfileHeaderWrapper - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A977
    - Requires none ✅ 
- [ ] ProfileHeader - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A978
    - Requires ProfileHeaderWrapper,  UserBadge (for *Follows you*), Button, UserAvatar/SingleUser ❌
- [x] ProfileTabs - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A989
    - Requires none ✅
- [x] UserBadgeLg - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A990
    - Requires none ✅
- [x] ProfileAbout - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A1048
    - Requires UserBadgeLg ❌
- [x] BaseDropdownSm
    - Includes:
       - CalendarDropdown - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A1049
       -  ShareDropdown - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A1054
    - Requires none ✅
- [x] UpcomingRoomCardLg - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=8277%3A1062
    - Requires CalendarDropdown, ShareDropdown, UserAvatar/MultipleUsers, Tag ❌

---

Guidelines:
- Don't add margin around the components.
- The bordered rectangle around each component in Figma is just for delimitation purposes.
- Put your component in `kibbeh/src/ui` and create a story in `kibbeh/src/stories`.
- Use tailwind for styling (see the tailwind config for theme info).
- You have to use TypeScript.
- Work on the `staging` branch.
- Please comment below if you start to work on a component, but only claim a component if you plan to work on it right away, so it doesn't block other people.
- Don't start working on a component if its dependencies aren't done.

Mockup & grid: https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=4911%3A0

> Most icons should be available under `kibbeh/src/icons`, but if the one you need isn't, ask on Discord or place a temp one with a comment pointing it out

If you have any questions ask Ben or @ ajb on [Discord](https://discord.gg/82HzQCJCDg).

**Comment by Angshu31** [comment]
I'm claiming "Calendar Dropdown" and "Share Dropdown" because both are practically the same.

**Comment by BronzW** [comment]
I'll do `ProfileAbout` once @goldyydev 's `UserBadgeLg` is merged.

**Comment by ajmnz** [comment]
> I'm claiming "Calendar Dropdown" and "Share Dropdown" because both are practically the same.

Merged into `BaseDropdownSm`, check my edit

**Comment by goldyydev** [comment]
I'm on `UserBadgeLg`

**Comment by Angshu31** [comment]
> > I'm claiming "Calendar Dropdown" and "Share Dropdown" because both are practically the same.
> 
> Merged into `BaseDropdownSm`, check my edit

Just to clarify, This means that I'll only create "BaseDropdownSm" component right? (The component will accept children)

**Comment by ajmnz** [comment]
> > > I'm claiming "Calendar Dropdown" and "Share Dropdown" because both are practically the same.
> > 
> > 
> > Merged into `BaseDropdownSm`, check my edit
> 
> Just to clarify, This means that I'll only create "BaseDropdownSm" component right? (The component will accept children)

Yup

**Comment by goldyydev** [comment]
I'm done with `UserBadgeLg`, waiting on @benawad to fix Storybook so I can PR

**Comment by sahilmob** [comment]
I will do `ProfileHeaderWrapper`

**Comment by Mvrs** [comment]
Is anyone working on the profileAbout?

ill do `ProfileHeader`

**Comment by Mayowa-Ojo** [comment]
I'll do ProfileTabs

**Comment by sahilmob** [comment]
I will do `CalendarDropdown`

**Comment by Mvrs** [comment]
@goldyydev there is a storybook issue huh? I thought I was tripping out 😆.

**Comment by BronzW** [comment]
Storybook issue now fixed

**Comment by sahilmob** [comment]
I will do `UpcomingRoomCardLg`

**Comment by Mvrs** [comment]
@sahilmob one thing about the ProfileHeaderWrapper, I believe the width is too much. Check out my PR and look at the story: #2111

**Comment by sahilmob** [comment]
> @sahilmob one thing about the ProfileHeaderWrapper, I believe the width is too much. Check out my PR and look at the story: #2111

I believe `ProfileHeaderWrapper` width should be bounded by its parent width

**Comment by BronzW** [comment]
Any abandoned components yet?

**Comment by ajmnz** [comment]
New components available for the Settings page 👉🏻 #2174
